staging/src/k8s.io/api/auditregistration/v1alpha1/types.go

```diff
@@ -192,3 +219,107 @@ type ServiceReference struct {
 	// +optional
 	Path *string `json:"path,omitempty" protobuf:"bytes,3,opt,name=path"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AuditClass is a set of rules that categorize requests
+//
+// This should be considered a highly privileged object, as modifying it
+// will change what is logged.
+type AuditClass struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec is the spec for the audit class
+	Spec AuditClassSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AuditClassList is a list of AuditClass items
+type AuditClassList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// List of audit classes
+	Items []AuditClass `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// AuditClassSpec is the spec for the audit class
+type AuditClassSpec struct {
+	// RequestSelectors defines a list of RequestSelectors
+	RequestSelectors []RequestSelector `json:"requestSelectors,omitempty" protobuf:"bytes,1,opt,name=requestSelectors"`
+}
+
+// RequestSelector selects requests by matching on the given fields. Selectors are
+// used to compose audit classes.
+type RequestSelector struct {
+	// The users (by authenticated user name) in this attribute group.
+	// An empty list implies every user.
+	// +optional
+	Users []string `json:"users,omitempty" protobuf:"bytes,2,rep,name=users"`
+	// The user groups in this attribute group. A user is considered matching
+	// if it is a member of any of the UserGroups.
+	// An empty list implies every user group.
```

Comment:
This doesn't seem right at all-- if you neglect to set this field, the users field is pointless, as every user is a member of some group (e.g., authenticated).

Reply:
@lavalamp This is more or less copy of the current documentation and behavior of [AuditPolicy](https://github.com/kubernetes/apiserver/blob/fbed0ffc300f09cdb8414c1dab89b28184696f33/pkg/apis/audit/v1/types.go#L188-L196).